rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/AdjunctResource.java

```diff
@@ -0,0 +1,264 @@
+
+        return AdjunctTransformer.adjunctDetail(brooklyn(), entity, adjunct, ui.getBaseUriBuilder());
+    }
+    
+    @Override
+    public Status getStatus(String application, String entityToken, String adjunctId) {
+        return AdjunctTransformer.inferStatus( brooklyn().getAdjunct(application, entityToken, adjunctId) );
+    }
+
+    @Override
+    public Response start(String application, String entityToken, String adjunctId) {
+        EntityAdjunct adjunct = brooklyn().getAdjunct(application, entityToken, adjunctId);
+        if (adjunct instanceof Policy) ((Policy)adjunct).resume();
+        else if (adjunct instanceof Feed) ((Feed)adjunct).resume();
+        else throw WebResourceUtils.badRequest("%s does not support start/resume", adjunct);
+        
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+
+    @Override
+    public Response stop(String application, String entityToken, String adjunctId) {
+        EntityAdjunct adjunct = brooklyn().getAdjunct(application, entityToken, adjunctId);
+        if (adjunct instanceof Policy) ((Policy)adjunct).suspend();
+        else if (adjunct instanceof Feed) ((Feed)adjunct).suspend();
+        else throw WebResourceUtils.badRequest("%s does not support suspend", adjunct);
+        
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+
+    @Override
+    public Response destroy(String application, String entityToken, String adjunctId) {
+        Entity entity = brooklyn().getEntity(application, entityToken);
+        EntityAdjunct adjunct = brooklyn().getAdjunct(entity, adjunctId);
+
+        if (adjunct instanceof Policy) {
+            ((Policy)adjunct).suspend();
+            entity.policies().remove((Policy) adjunct);
+        } else if (adjunct instanceof Enricher) {
+            entity.enrichers().remove((Enricher) adjunct);
+        } else if (adjunct instanceof Feed) {
+            ((Feed)adjunct).suspend();
+            ((EntityInternal)entity).feeds().remove((Feed) adjunct);
+        } else {
+            throw WebResourceUtils.badRequest("Unexpected adjunct type %s", adjunct);
+        }
+        
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+    
+    // ---- config ----
+    
+    @Override
+    public List<ConfigSummary> listConfig(
+            final String application, final String entityToken, final String adjunctToken) {
+        Entity entity = brooklyn().getEntity(application, entityToken);
+        EntityAdjunct adjunct = brooklyn().getAdjunct(entity, adjunctToken);
+
+        List<ConfigSummary> result = Lists.newArrayList();
+        for (ConfigKey<?> key : adjunct.config().findKeysPresent(Predicates.alwaysTrue())) {
+            result.add(ConfigTransformer.of(key).on(entity, adjunct).includeLinks(ui.getBaseUriBuilder(), false, true).transform());
+        }
+        return result;
+    }
+
+    // TODO support parameters  ?show=value,summary&name=xxx &format={string,json,xml}
```

Comment:
I'd move that comment to the `AdjunctsApi` interface, rather than in the impl (given it's about the api rather than the implementation).

Reply:
already there